src/BuiltinExtensions/ComfyUIBackend/ExtraNodes/SwarmComfyCommon/SwarmSaveImageWS.py

```diff
@@ -22,19 +24,46 @@ def INPUT_TYPES(s):
     OUTPUT_NODE = True
     DESCRIPTION = "Acts like a special version of 'SaveImage' that doesn't actual save to disk, instead it sends directly over websocket. This is intended so that SwarmUI can save the image itself rather than having Comfy's Core save it."
 
-    def save_images(self, images):
+    def save_images(self, images, bit_depth):
         pbar = comfy.utils.ProgressBar(SPECIAL_ID)
         step = 0
         for image in images:
-            i = 255.0 * image.cpu().numpy()
-            img = Image.fromarray(np.clip(i, 0, 255).astype(np.uint8))
+            if bit_depth == "16":
+                img_np = image.cpu().numpy()
+
+                if img_np.ndim != 3 or img_np.shape[2] != 3:
```

Comment:
why

Reply:
That checks if the ndim (number of dimensions in an image: height, width, and color channels) is not three (not a standardad RGB image) or if the shape of the third dimension (color channels) is not three (not an RGB).
This prevents the erroneous processing of things like:
Grayscale images (2D arrays)
Images with alpha channels (4 channels instead of 3)
Any other non-standard image format

Considering that we will be receiving this from comfyui, we could assume that we will always receive good data, and forego this check. A grayscale image generated by any checkpoint would still be 8bit, so this is might be redundant. I'm happy to take this out if you also think so. The reason this came to be was that I was testing this script locally first, and the images that I had picked to test turned out to be buggy (synthetically generated by another application for a different purpose). This helped me suss that out and get back on the right path. Different situations there.

Reply:
But why is this check needed for 16bit depth, but not 8bit? If the 8bit code handles nonstandard image formats, 16bit should too. if the 8bit code can't, the check should be outside the if block.

Reply:
Really good point. Didn't think of that. Will implement!